src/tools/recvfuzz/recvfuzz.cpp

```diff
@@ -0,0 +1,681 @@
+/*++
+
+    Copyright (c) Microsoft Corporation.
+    Licensed under the MIT License.
+
+Abstract: 
+    Packet Fuzzer tool in the receive path.
+--*/
+
+#include <time.h>
+#include <stdio.h>
+
+#include <vector>
+#include <map>
+#include <mutex>
+#include <algorithm>
+
+
+#define QUIC_TEST_APIS 1 // Needed for self signed cert API
+#define QUIC_API_ENABLE_INSECURE_FEATURES 1 // Needed for disabling 1-RTT encryption
+#ifndef NOMINMAX
+#define NOMINMAX
+#endif
+#include "precomp.h" // from core directory
+#include "msquichelper.h"
+#include "msquic.hpp"
+
+const MsQuicApi* MsQuic;
+#define MagicCid 0x989898989898989ull
+const QUIC_HKDF_LABELS HkdfLabels = { "quic key", "quic iv", "quic hp", "quic ku" };
+uint64_t RunTimeMs;
+
+#define ASSERT_ON_FAILURE(x) \
+    do { \
+        QUIC_STATUS _STATUS; \
+        CXPLAT_FRE_ASSERT(QUIC_SUCCEEDED((_STATUS = x))); \
+    } while (0)
+#define ASSERT_ON_NOT(x) CXPLAT_FRE_ASSERT(x)
+
+static const char* Alpn = "fuzz";
+static uint32_t Version = QUIC_VERSION_DRAFT_29;
+const char* Sni = "localhost";
+
+#define QUIC_MIN_INITIAL_LENGTH 1200
+
+struct StrBuffer
+{
+    uint8_t* Data;
+    uint16_t Length;
+
+    StrBuffer(const char* HexBytes)
+    {
+        Length = (uint16_t)(strlen(HexBytes) / 2);
+        Data = new uint8_t[Length];
+
+        for (uint16_t i = 0; i < Length; ++i) {
+            Data[i] =
+                (DecodeHexChar(HexBytes[i * 2]) << 4) |
+                DecodeHexChar(HexBytes[i * 2 + 1]);
+        }
+    }
+
+    ~StrBuffer() { delete [] Data; }
+};
+
+
+class FuzzingData {
+    const uint8_t* data;
+    size_t size;
+    std::mutex mux;
+    // TODO: support bit level pointers
+    size_t Ptrs;
+    size_t NumIterated;
+    bool Cyclic;
+
+    bool CheckBoundary(size_t Adding) {
+        // TODO: efficient cyclic access
+        if (size < Ptrs + Adding) {
+            if (!Cyclic) {
```

Comment:
`Cyclic` is **never** `false` so it might be worth refactoring to remove the variable completely?